digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java

```diff
@@ -324,27 +324,34 @@ protected TaskResult callExecutor(Path projectPath, String type, TaskRequest mer
         return operator.run();
     }
 
-    private static class OperatorSecretFilter
-            implements Predicate<String>
+    private static class AgentOperatorSecretFilter
+            implements DefaultSecretProvider.OperatorSecretFilter
```

Comment:
Does this class need to have `allowUserSecretAccess` method? It seems there is no actual usage of this class without calling the method. If we can remove the method, this class can be simpler like this:
```
    private static class AgentOperatorSecretFilter
            implements DefaultSecretProvider.OperatorSecretFilter
    {
        private final Set<String> predeclaredSecretKeys;
        private final Predicate<String> dynamicFilter;

        AgentOperatorSecretFilter(OperatorFactory factory, Operator operator)
        {
            this.predeclaredSecretKeys = ImmutableSet.copyOf(
                    factory.getSecretAccessList().getSecretKeys());
            this.dynamicFilter = (key) -> operator.testUserSecretAccess(key);
        }

        @Override
        public boolean test(String key, boolean userGranted)
        {
            if (userGranted || predeclaredSecretKeys.contains(key)) {
                return dynamicFilter.test(key);
            }
            return false;
        }
    }
```